### PR TITLE
[stdlib] Migrate stdlib tests of Swift 3

### PR DESCRIPTION
**Pull request by airspeedswift**
PR to gather std lib tests that use Swift 3.

- Some need switching from 3 to 4, with source tweaks
- Some just need deleting because they were testing 3-only stuff
- Some test multiple versions and need 3 removing

Feel free to join the party.

### CODE REVIEW

**Comment by airspeedswift** [comment]
@swift-ci please test

File: test/stdlib/Dispatch.swift [hunk 1]
```diff
@@ -595,11 +595,9 @@ DispatchAPI.test("DispatchTimeInterval") {
 	expectTrue(t == t) // This would crash.
 }
 
-#if swift(>=4.0)
```
Comment (slavapestov) [inline test/stdlib/Dispatch.swift]: Can you remove the RUN: line for -swift-version 3 too?

File: test/stdlib/LazyCollectionPlus.swift [hunk 2]
```diff
@@ -1,5 +1,5 @@
 // RUN: %empty-directory(%t)
-// RUN: %target-build-swift %s -o %t/a.out3 -swift-version 3 && %target-run %t/a.out3
+// RUN: %target-build-swift %s -o %t/a.out -swift-version 3 && %target-run %t/a.out
```
Comment (slavapestov) [inline test/stdlib/LazyCollectionPlus.swift]: Should this be -swift-version 4?

**Comment by slavapestov** [comment]
Thanks for working on this!

**Comment by milseman** [comment]
Some of these changes are to explicitly check legacy support for users of -swift-version 3. Is your intention for us to not test this mode at all, or just to migrate stale tests?

**Review by milseman — APPROVED** [review]
LGTM

File: test/stdlib/StringFlatMap.swift [hunk 3]
```diff
@@ -1,23 +1,14 @@
...
 // RUN: %target-build-swift %s -o %t/a.out4 -swift-version 4 && %target-run %t/a.out4
 
 // REQUIRES: executable_test
 
 import StdlibUnittest
 
-#if swift(>=4)
 
 public typealias ExpectedResultType = [Character]
 let swiftVersion = "4"
```
Comment (lorentey) [inline test/stdlib/StringFlatMap.swift]: Nit: I think we should remove these two variables now; they make it slightly harder to understand the tests.

**Review by lorentey — APPROVED** [review]
Big thumbs up!

File: test/stdlib/LazyCollectionPlus.swift (hunk 2 above)
Comment (airspeedswift) [reply test/stdlib/LazyCollectionPlus.swift]: ah in fact the whole test should just go, we don't support this beyond Swift 3

**Comment by airspeedswift** [comment]
@swift-ci please test

**Comment by airspeedswift** [comment]
@swift-ci please test Linux platform

**Comment by airspeedswift** [comment]
@swift-ci please test Linux platform

**Comment by airspeedswift** [comment]
@swift-ci please clean test linux platform

**Comment by airspeedswift** [comment]
@swift-ci please test

File: test/stdlib/Inputs/DictionaryKeyValueTypesObjC.swift [hunk 4]
```diff
@@ -475,8 +475,7 @@ func getBridgedNSDictionaryOfRefTypesBridgedVerbatim() -> NSDictionary {
   d[TestObjCKeyTy(20)] = TestObjCValueTy(1020)
   d[TestObjCKeyTy(30)] = TestObjCValueTy(1030)
 
-  let bridged =
-    unsafeBitCast(convertDictionaryToNSDictionary(d), to: NSDictionary.self)
```
Comment (airspeedswift) [inline test/stdlib/Inputs/DictionaryKeyValueTypesObjC.swift]: @lorentey just to check, can you spot any higher purpose to this bit cast that I'm missing? The function already returns an `NSDictionary`.

**Comment by airspeedswift** [comment]
@swift-ci please test linux platform

**Comment by airspeedswift** [comment]
@swift-ci please test Linux platform

**Comment by airspeedswift** [comment]
@swift-ci please test Linux platform

**Comment by airspeedswift** [comment]
Finally™️

File: test/stdlib/Inputs/DictionaryKeyValueTypesObjC.swift (hunk 4 above)
Comment (lorentey) [reply test/stdlib/Inputs/DictionaryKeyValueTypesObjC.swift]: There is none as far as I can tell!

**Pull request merged**